ddl/ddl_api.go

```diff
@@ -2779,6 +2851,37 @@ func checkColumnWithIndexConstraint(tbInfo *model.TableInfo, originalCol, newCol
 	return nil
 }
 
+func checkAutoRandom(tableInfo *model.TableInfo, originCol *table.Column, specNewColumn *ast.ColumnDef) error {
+	if !config.AutoRandom.IsAllow() && containsColumnOption(specNewColumn, ast.ColumnOptionAutoRandom) {
+		return ErrInvalidAutoRandom.GenWithStackByArgs(autoid.AutoRandomExperimentalDisabledErrMsg)
+	}
+	// Disallow add/drop/modify actions on auto_random.
+	newAutoRandomBit := getAutoRandomBits(specNewColumn)
+	uNewAutoRandomBit, err := convertAutoRandomBitsToUnsigned(newAutoRandomBit)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if tableInfo.AutoRandomBits != uNewAutoRandomBit {
```

Comment:
A question: is there a priority between `tableInfo.AutoRandomBits` and `column specified RandomBits`.

Reply:
There is only one `AutoRandomBits`. The `ColumnOptionAutoRandom` in AST is finally stored in the tableInfo.AutoRandomBits.

Reply:
got it, the definition is on `column`, but the store location is on `tableInfo`.